Exdi/exdigdbsrvsample/Start-ExdiDebugger.ps1

```diff
@@ -0,0 +1,314 @@
+﻿<#
+.Synopsis
+  Installs and launches exdi debugger (automating xml file editing)
+
+.Description
+  This script will install ExdiGdbSrvSample.dll if required, configure the xml settings
+  files, check for running dllhost.exe processes, and launch the debugger to connect to
+  an already running gdb server hardware debugging target.
+
+.Parameter ExdiTarget
+    Type of target to connect to. This corresponds to a specific section in the settings xml file
+
+.Parameter HostName
+    IP address or hostname of the computer hosting the gdb server session (defaults to "LocalHost")
+
+.Parameter GdbPort
+    Port that the gdb server is listening on.
+
+.Parameter Architecture
+    Architecture of the hardware debugging target (this parameter also implies the ArchitectureFamily
+    parameter in the xml settings file)
+
+.Parameter ExdiDropPath
+    Location of the ExdiGdbSrvSample.dll, exdiConfigData.xml, and systemregisters.xml files. These will
+    only be copied if ExdiGdbSrvSample.dll is not installed or is installed incorrectly.
+
+.Parameter ExtraDebuggerArgs
+    Extra arguments to pass on the debugger command line
+
+.Parameter PreNTAppDebugging
+    Changes the value of the heuristicScanSize from 0xfffe (best for NT) to 0xffe (for pre-NT Apps)
+
+.Parameter DontTryDllHostCleanup
+    Checking for existing running instances of ExdiGdbSrvSample.dll in dllhost.exe requires elevation.
+    Providing this switch will allow the script to run without elevation (although the debugger may not
+    function correctly).
+
+.Example
+    >---------------- MSFT External usage (first run) ------------<
+    .\Start-ExdiDebugger.ps1 -ExdiTarget "QEMU" -GdbPort 1234 -Architecture x64 -ExdiDropPath "C:\path\to\built\exdi\files"
+
+.Example
+    >---------------- MSFT Internal usage (first run) ------------<
+    .\Start-ExdiDebugger.ps1 -ExdiTarget "QEMU" -GdbPort 1234 -Architecture x64 -ExdiDropPath "\dbg\privates\daily\latest\uncompressed\amd64\exdi"
+
+.Example
+    PS>.\Start-ExdiDebugger.ps1 -ExdiTarget "QEMU" -GdbPort 1234 -Architecture x64
+#>
+
+[CmdletBinding()]
+param
+(
+    [ValidateSet("QEMU")]
+    [string]
+    $ExdiTarget = "QEMU",
+
+    [string]
+    $HostName = "LocalHost",
+
+    [Parameter(Mandatory=$true)]
+    [Int]
+    $GdbPort,
+
+    [Parameter(Mandatory=$true)]
+    [string]
+    [ValidateSet("x86", "x64", "arm64")]
+    $Architecture,
+
+    [string]
+    $ExdiDropPath,
+
+    [string]
+    $DebuggerPath = "$env:LOCALAPPDATA\DBG\UI\WindbgX.exe",
+
+    [string[]]
+    $ExtraDebuggerArgs = @(),
+
+    [switch]
+    $PreNTAppDebugging,
+
+    [switch]
+    $DontTryDllHostCleanup
+)
+
+$ErrorActionPreference = "Stop"
+
+#region Functions
+
+Function Test-Admin
+{
+    ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]“Administrator”)
+}
+
+Function Test-ParameterValidation
```

Comment:
Nit: This function does more than parameter validation. It also determines the debugger path it seems, right?